shell/platform/android/io/flutter/embedding/android/FlutterFragmentActivity.java

```diff
@@ -710,4 +716,34 @@ protected RenderMode getRenderMode() {
   private boolean isDebuggable() {
     return (getApplicationInfo().flags & ApplicationInfo.FLAG_DEBUGGABLE) != 0;
   }
+
+  /**
+   * Registers all plugins that an app lists in its pubspec.yaml.
+   *
+   * <p>The Flutter tool generates a class called GeneratedPluginRegistrant, which includes the code
+   * necessary to register every plugin in the pubspec.yaml with a given {@code FlutterEngine}. The
+   * GeneratedPluginRegistrant must be generated per app, because each app uses different sets of
+   * plugins. Therefore, the Android embedding cannot place a compile-time dependency on this
+   * generated class. This method uses reflection to attempt to locate the generated file and then
+   * use it at runtime.
+   *
+   * <p>This method fizzles if the GeneratedPluginRegistrant cannot be found or invoked. This
+   * situation should never occur, but if any eventuality comes up that prevents an app from using
+   * this behavior, that app can still write code that explicitly registers plugins.
+   */
+  private static void registerPlugins(@NonNull FlutterEngine flutterEngine) {
```

Comment:
Looks like this is duplicated from `FlutterActivity`.

Can the `FlutterActivity` method be made package protected and invoked here?  Or maybe this should be a utility method somewhere in the `io.flutter.embedding.engine.plugins` package?

Reply:
oh ya, good idea. Done.